libbeat/docs/troubleshooting/metrics-in-logs.asciidoc

```diff
@@ -0,0 +1,105 @@
+| `.filebeat`            | Object | Information specific to Filebeat, e.g. harvester, events |
+|===
+
+[cols="1,1,2,2"]
+|===
+| Field path (relative to `.monitoring.metrics.beat`) | Type    | Meaning                              | Troubleshooting hints
+
+| `.runtime.goroutines` | Integer | Number of goroutines running | If this number grows over time, it indicates a goroutine leak
+|===
+
+[cols="1,1,2,2"]
+|===
+| Field path (relative to `.monitoring.metrics.libbeat`) | Type    | Meaning                              | Troubleshooting hints
+
+| `.pipeline.events.active` | Integer | Number of events currently in the libbeat publisher pipeline. | If this number grows over time, it may indicate that the Beat (e.g. Filebeat) is producing events faster than the output can consume. Consider increasing the number of output workers (if this setting is supported by the output; Elasticsearch and Logstash outputs support this setting). If this number reaches the maximum queue size (`queue.mem.events` for the in-memory queue), it may indicate backpressure on the Beat, implying that the Beat may need to stop ingesting more events from the source.
+| `.output.events.total` | Integer | Number of events currently being processed by the output. | If this number grows over time, it may indicate that the output destination (e.g. Logstash pipeline or Elasticsearch cluster) is not being able to keep up with accepting events at the same or faster rate than what the Beat is sending to it.
+| `.output.events.acked` | Integer | Number of events acknowledged by the output destination. | Generally, we want this number to be the same as `.output.events.total` as this indicates that the output destination has reliably received all the events sent to it.
+| `.output.events.failed` | Integer | Number of events that the Beat tried to send to the output destination but it failed to receive them. | Generally, we want this field to be absent or its value to be 0. When the value is greater than zero, it's useful to check the Beat's logs right before this log entry's `@timestamp` to see if there are any connectivity issues with the output destination. Note that failed events are not lost or dropped; they will be sent back to the publisher pipeline for retrying later.
+|===
+
+[cols="1,1,2,2"]
+|===
+| Field path (relative to `.monitoring.metrics.filebeat`) | Type    | Meaning                              | Troubleshooting hints
+
+| `.events.active` | Integer | Number of events being actively processed by Filebeat (including events Filebeat has already sent to the libbeat publisher pipeline, but not including events the pipeline has sent to the output). | If this number grows over time, it may indicate that Filebeat inputs are harvesting events too fast for the pipeline and output to keep up.
+|===
+
+=== Useful commands
+
+==== Check if Filebeat is processing events
```

Comment:
This section only makes sense for Filebeat.  Is there some way to exclude this section from other Beats' documentation?

Reply:
Fixed via 252e92b